robosherlock/descriptors/annotators/filter/PointCloudFilter.yaml

```diff
@@ -3,7 +3,8 @@ annotator:
   implementation: rs_PointCloudFilter
   description: 'Filters a point cloud given X,Y,and Z limits while keeping it organized.'
 parameters:
-  transform_cloud: false
+  source_frame: 'head_rgbd_sensor_rgb_frame'
```

Comment:
what about leaving the transform cloud flag? If true it transforms cloud into target frame. If false filtering happens on the original point cloud; 

source_frame is not needed. Should be found in ``cloud->header.frame_id`` (If this frame_id is not set we have to make sure ImagePreprocessor sets it to the correct value); This will make it a bit more independent of the camera. Users  will only have to set the target frame.

Reply:
frame_id is indeed not set. I leave the source_frame parameter until this is fixed

Reply:
@SirrahErydya yeah, I thought so. you can modify  `src/utils/src/ImagePreprocessor.cpp`, namely the `createCloud `  method to do the following after it projects the color image onto the depth:

```
      rs::DepthImageProcessing::project(depth, color, alpha, lookupX, lookupY, cloud);
      sensor_msgs::CameraInfo camInfo;      
      cas.get(VIEW_CAMERA_INFO, camInfo)
      cloud->header.frame_id = camInfo.header.frame_id;
      cloud->header.stamp = camInfo.header.stamp.toNSec();
```